File: solutions/typescript/bob/1/bob.ts
export function hey(message: string): string {
    message = message.trim()
    if (message === '') {
        return 'Fine. Be that way!'
    }
    if (!message.match(/[a-z]/) && message.match(/[A-Z]/)) {
        if (message.endsWith('?')) {
            return "Calm down, I know what I'm doing!"
        }
        return 'Whoa, chill out!'
    }
    if (message.endsWith('?')) {
        return 'Sure.'
    }
    return 'Whatever.'
}

File: solutions/lua/bob/1/bob.lua
local bob = {}

function bob.hey(say)
    say = say:gsub('^%s*(.-)%s*$', '%1')
    if say == '' then
        return 'Fine, be that way.'
    end
    if not say:match('[a-z]') and say:match('[A-Z]') then
        if (say:sub(-1) == '?') then
            return 'Calm down, I know what I\'m doing!'
        end
        return 'Whoa, chill out!'
    end
    if (say:sub(-1) == '?') then
        return 'Sure'
    end
    return 'Whatever'
end

return bob
